new-components/reporters/defectdojo/internal/reporter/reporter.go

```diff
@@ -0,0 +1,321 @@
+package reporter
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"log/slog"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/go-errors/errors"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/smithy-security/pkg/env"
+
+	"github.com/smithy-security/smithy/new-components/reporters/defectdojo/internal/client"
+	"github.com/smithy-security/smithy/sdk/component"
+	vf "github.com/smithy-security/smithy/sdk/component/vulnerability-finding"
+	ocsffindinginfo "github.com/smithy-security/smithy/sdk/gen/ocsf_ext/finding_info/v1"
+	ocsf "github.com/smithy-security/smithy/sdk/gen/ocsf_schema/v1"
+)
+
+const (
+	// DojoTimeFormat is the time format accepted by defect dojo.
+	dojoTimeFormat = "2006-01-02"
+
+	// DojoTestTimeFormat is the time format expected by defect dojo when creating a test.
+	dojoTestTimeFormat = "2006-01-02T03:03"
+)
+
+type (
+	Conf struct {
+		User       string
+		Token      string
+		URL        string
+		ProductID  int32
+		Template   string
+		instanceID string
+	}
+	dojoLogger struct {
+		conf   *Conf
+		client *client.Client
+	}
+)
+
+// NewConf returns a new configuration build from environment lookup.
+func NewConf(envLoader env.Loader) (*Conf, error) {
+	var envOpts = make([]env.ParseOption, 0)
+	if envLoader != nil {
+		envOpts = append(envOpts, env.WithLoader(envLoader))
+	}
+
+	instanceId, err := env.GetOrDefault(
+		"SMITHY_INSTANCE_ID",
+		"",
+		append(envOpts, env.WithDefaultOnError(false))...,
+	)
+	if err != nil {
+		return nil, errors.Errorf("could not get SMITHY_INSTANCE_ID: %w", err)
+	}
+
+	dojoUser, err := env.GetOrDefault(
+		"DOJO_USER",
+		"administrator",
+		append(envOpts, env.WithDefaultOnError(false))...,
+	)
+	if err != nil {
+		return nil, errors.Errorf("could not get env variable for DOJO_USER: %w", err)
+	}
+
+	dojoKey, err := env.GetOrDefault(
+		"DOJO_API_KEY",
+		"",
+		append(envOpts, env.WithDefaultOnError(false))...,
+	)
+	if err != nil {
+		return nil, errors.Errorf("could not get env variable for DOJO_API_KEY: %w", err)
+	}
+
+	dojoURL, err := env.GetOrDefault(
+		"DOJO_API_URL",
+		"",
+		append(envOpts, env.WithDefaultOnError(false))...,
```

Comment:
If `env.WithDefaultOnError(false)` is used we can simply omit it as we return error by default unless `env.WithDefaultOnError(true)` is used

Reply:
what is "it"? can we omit `env.WithDefaultOnError(false)`?

Reply:
yep